src/clang.rs

```diff
@@ -387,16 +387,26 @@ impl Cursor {
 
     /// Given that this cursor's referent is a function, return cursors to its
     /// parameters.
-    pub fn args(&self) -> Vec<Cursor> {
+    pub fn args(&self) -> Option<Vec<Cursor>> {
+        // XXX: We might want to use and keep num_args
```

Comment:
Returning an iterator is definitely preferable, since it's more efficient. You can implement something like `TypeTemplateArgIterator` (in that same file). That being said, only a `Vec` is also fine if you prefer :)